eZ/Publish/Core/Persistence/Legacy/Content/Type/Gateway/DoctrineDatabase.php

```diff
@@ -397,7 +396,7 @@ protected function setCommonTypeColumns(Query $q, ValueObject $type)
         )->set(
             $this->dbHandler->quoteColumn('language_mask'),
             $q->bindValue(
-                $this->languageMaskGenerator->generateLanguageMask($type->name),
+                $this->languageMaskGenerator->generateLanguageMaskFromLanguageCodes($type->languageCodes, array_key_exists('always-available', $type->name)),
```

Comment:
judging from mask generator code, always avaiale flag is boolean, so maybe rather: `$type->name['always-available'] ?? false`

ref:
`$mask |= $languages['always-available'] ? 1 : 0;`

Reply:
Yes, but there are some tests that have following fixture, so not sure if that is completely safe.
```
        $type->name = array(
            'always-available' => 'eng-US',
            'eng-US' => 'Folder',
            'eng-GB' => 'Folder (GB)',
        );
```

Reply:
I also noticed that, but it could be it expects string or false, as the code I pasted uses a loose check.

Reply:
Make sense :+1:

Reply:
I would rather change it to `$type->name['always-available'] !== false`
to exclude possible `$type->name['always-available'] = false`. as with `$type->name['always-available'] ?? false` we could pass string or bool into `generateLanguageMaskFromLanguageCodes`. Unless I am reading your point somehow wrong.

Reply:
> I would rather change it to `$type->name['always-available'] !== false`
> to exclude possible `$type->name['always-available'] = false`. as with `$type->name['always-available'] ?? false` we could pass string or bool into `generateLanguageMaskFromLanguageCodes`.

`$type->name['always-available'] !== false` will crash if the key isn't set.

@andrerom this API is a little bit weird, `always-available` stores not boolean, but a language code pointing to an always available language code. Calling `generateLanguageMaskFromLanguageCodes` with null-coalescing operator would crash as the 2nd argument is required to be boolean. 
So I actually prefer here what @ViniTou did.